src/test/java/org/elasticsearch/test/rest/section/SkipSection.java

```diff
@@ -57,27 +74,39 @@ public String getReason() {
         return reason;
     }
 
-    public boolean skip(String currentVersion) {
+    public boolean skip(Version currentVersion) {
         if (isEmpty()) {
             return false;
         }
-
-        if (version != null) {
-            return VersionUtils.skipCurrentVersion(version, currentVersion);
-        }
-
-        if (features != null && !this.features.isEmpty()) {
-            return !Features.areAllSupported(this.features);
+        if (isVersionCheck()) {
+            return currentVersion.onOrAfter(lowerVersion) && currentVersion.onOrBefore(upperVersion);
+        } else {
+            return Features.areAllSupported(features) == false;
         }
-
-        throw new IllegalArgumentException("version or feature should be not null in a non empty skip section");
     }
 
     public boolean isVersionCheck() {
-        return Strings.hasLength(version);
+        return features.isEmpty();
     }
 
     public boolean isEmpty() {
         return EMPTY.equals(this);
     }
+    
+    private Version[] parseVersionRange(String versionRange) {
```

Comment:
as I said above where I summoned Clint (I wasn't sure yet if we had changed the parsing logic), if we change how we parse REST tests, all of the clients tests infra need to adapt too. Can we avoid breaking bw comp here and leave the previous syntax?